tests/unit/phpDocumentor/Plugin/Standards/phpDocumentor/Sniff/ArgumentHasParamTagTest.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * phpDocumentor
+ *
+ * PHP Version 5.3
+ *
+ * @copyright 2010-2014 Mike van Riel / Naenius (http://www.naenius.com)
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor\Plugin\Standards\phpDocumentor\Sniff;
+
+use Mockery as m;
+
+/**
+ * Tests for the ArgumentHasParamTag class.
+ */
+class ArgumentHasParamTagTest extends \PHPUnit_Framework_TestCase
+{
+    const EXAMPLE_RULE_NAME  = 'my.rule.name';
+    const EXAMPLE_DESCRIPTOR = 'File';
+
+    /** @var m\MockInterface */
+    private $validatorMock;
+
+    /** @var ArgumentHasParamTag */
+    private $fixture;
+
+    /**
+     * Initializes the fixture and its dependencies with mocks.
+     */
+    protected function setUp()
+    {
+        $this->validatorMock = m::mock('Symfony\Component\Validator\Validator');
```

Comment:
you might want to change all the hardcoded class strings to e.g. ```validator::class``` to allow code navigation and see usage statistics